src/librbd/image/OpenRequest.cc

```diff
@@ -574,6 +575,15 @@ template <typename I>
 Context *OpenRequest<I>::send_set_snap(int *result) {
   if (m_image_ctx->snap_name.empty() &&
       m_image_ctx->open_snap_id == CEPH_NOSNAP) {
+
+    auto io_scheduler_cfg =
```

Comment:
Nit: maybe move to to a new ```Context* finalize(int r)``` method to be prepared for the future when reads could be aggregated.